[DOC-11789] Vector Search - The Server Files

sarahlwelton opened the pull request:
This PR contains all of the documentation changes required for Vector Search on Couchbase Server. 

This PR will remain in draft until more review comments have been received. 

View the contents of the PR in context here: https://preview.docs-test.couchbase.com/sarah-AV-68571/server/current/vector-search/vector-search.html

Username and password is the same as for all docs preview sites: https://couchbasecloud.atlassian.net/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords

sarahlwelton committed:
[DOC-11667] Adding import-search-index.adoc

Fixing typos in create-search-index-ui

Adding content and fixing typos (apparently?) in create-search-indexes

Updating nav.adoc with new versions and new entries.

sarahlwelton committed:
Merge branch 'release/7.6' of https://github.com/couchbaselabs/docs-devex into DOC-11667-import-search-index-def

sarahlwelton committed:
[DOC-11667] Updating broken link in import-search-index

Moving import-search-index in nav.adoc

Adding link to import-search-index to create-search-index-ui

sarahlwelton committed:
[DOC-11668] Adding note about index names to:

create-quick-index.adoc
create-search-index-ui.adoc
create-search-indexes.adoc
create-search-index-rest-api.adoc

sarahlwelton committed:
Merge branch 'release/7.6' of https://github.com/couchbaselabs/docs-devex into DOC-11668-search-index-name-scoping

sarahlwelton committed:
[DOC-11672] Starting to write view-index-details

Adding image of index details button

Adding view-index-details to nav.adoc

sarahlwelton committed:
[DOC-11672] Language and formatting tweaks in view-index-details

sarahlwelton committed:
[DOC-11671] Initial edits.

Changing links in searchfun.adoc and adding mention of analyzer restriction changes

Adding anchor link to default-analyzers-reference

Adding note about how to find UUID to search-index-params

Adding notes about analyzer restriction changes to search-request-params

sarahlwelton committed:
Merge branch 'DOC-11671-search-n1ql-func-updates' into DOC-11666-7-6-search

sarahlwelton committed:
Fixing broken link

sarahlwelton committed:
Merge branch 'DOC-11672-index-details-page' into DOC-11666-7-6-search

sarahlwelton committed:
Merge branch 'DOC-11668-search-index-name-scoping' into DOC-11666-7-6-search

sarahlwelton committed:
Merge branch 'DOC-11667-import-search-index-def' into DOC-11666-7-6-search

sarahlwelton committed:
[DOC-11668] Updating note formatting in:

create-quick-index
create-search-index-rest-api
create-search-index-ui
create-search-indexes

sarahlwelton committed:
Merge branch 'DOC-11668-search-index-name-scoping' into DOC-11666-7-6-search

sarahlwelton committed:
[DOC-11667] Change location of links to info on Search indexes and aliases

sarahlwelton committed:
Merge branch 'DOC-11667-import-search-index-def' into DOC-11666-7-6-search

sarahlwelton committed:
[DOC-11666] Reviewing topics for polishing.

Changed NOTE in create-quick-index to block style admonition

Changed NOTE in create-search-index-rest-api to block style admonition

Changed NOTE in create-search-index-ui to block style admonition

Changed NOTE in create-search-indexes to block style admonition

Fixing negative contraction in search-index-params

Moving location of view-index-details in nav.adoc

sarahlwelton committed:
[DOC-11668] Updating code samples with scoped endpoints

sarahlwelton committed:
[DOC-11666] Tweaking type mapping descriptions to match Capella.

Small change to procedure in create-type-mapping.adoc

sarahlwelton committed:
[DOC-11666] Adjusting links in customize-index.adoc

sarahlwelton committed:
[AV-68962] First draft of vector search overview

Created nav.adoc partial for new vector search module.

Updated search.adoc to provide brief overview of Vector Search and link.

sarahlwelton committed:
[AV-68962] Changing indentation on nav.adoc

sarahlwelton committed:
[AV-68963] First draft of how to create a vector search index from the UI

sarahlwelton committed:
[AV-68962] Adding more notes about vector search.

sarahlwelton committed:
[DOC-11790] More language tweaks on overview

sarahlwelton committed:
[DOC-11791] Updating create-vector-search-index-ui.

Adding file to nav.adoc

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Created new file, child-field-options-reference

Updated create-child-field, quick-index-field-options, search-index-params, and search-request-params to support this change and add in vector search-specific changes

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Add reference for child fields to nav.adoc

Fix broken links in create-vector-search-index-ui.adoc

sarahlwelton committed:
[DOC-11790] Another language tweak

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11794] Initial draft of create and run vector search index topics for REST API

sarahlwelton committed:
[DOC-11794] Fixing typo in include, adding missing link

sarahlwelton committed:
[DOC-11794] Adding info on new field, vector_index_optimized_for

sarahlwelton committed:
[DOC-11791] Adding new field, Optimized For

sarahlwelton committed:
[DOC-11974] Adding useful anchor links to child-field-options-reference and search-index-params

Adding knn array to example json file

Adding documentation on knn array to search-request-params.adoc

sarahlwelton committed:
[DOC-11974] Adding notes about how k and size/limit duke it out

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11792] Add new file and example for running vector search from the UI

sarahlwelton committed:
[DOC-11792] Actually add new topic to nav.adoc

sarahlwelton committed:
[DOC-11790] Language tweak after chatting with SME

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Added Optimized For to create-vector-search-index-ui

Corrected description of "index this field as an identifier" option

Moved dimension, similarity metric, and optimized for descriptions into a partial to reuse across topics for consistency.

Added anchor link to "default-analyzers" to support updating description of "index field as identifier"

sarahlwelton committed:
[DOC-11791] Added Optimized For to create-vector-search-index-ui

Corrected description of "index this field as an identifier" option

Moved dimension, similarity metric, and optimized for descriptions into a partial to reuse across topics for consistency.

Added anchor link to "default-analyzers" to support updating description of "index field as identifier"

sarahlwelton committed:
[DOC-11794] Expanding page-toclevels to 2 for search-index-params.

Changing dims, similarity, and vector_index_optimized_for to partials for consistency.

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11794] Tweaking similarity description

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11975] Updating some links to go to the better docs

Adding a new example specific to searching a vector search index from the SEARCH function

sarahlwelton committed:
[DOC-11791] Applying some changes that got made on another branch

Increasing page-toclevels

Changing descriptions for vector search params to partials

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11790] Highlight vectors further up the page on search.adoc

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11790] Adding in links to other vector search topics

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11790] Adding line break

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11971] Adding missing links

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11794] Removing unnecessary comment

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Copying over some useful changes from the Capella side on type mapping explanations and GeoJSON data structure

sarahlwelton committed:
[DOC-11791] Correcting procedure for vector search index creation in Server UI.

Since it was stolen from geo-search-ui, fixing there, too.

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11789] Typo/consistency fix

sarahlwelton committed:
[DOC-11791] Adding vector to field data types (duh)

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11789] Consistency fix

sarahlwelton committed:
[DOC-11794] Adding missing direct anchor link

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11794] Clarifying k property behaviour

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11975] Fixing a missed link, adding a note about vector search example

sarahlwelton committed:
Merge branch 'DOC-11975-sqlpp-vector-search' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11790] Adjusting note to link to relevant documentation and explain what happens

sarahlwelton committed:
Merge branch 'DOC-11790-vector-search-overview' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Adding anchor link for "dimension" in child-field-options

Moving child-field-options-reference in the nav

Adding changes made for Advanced/Quick in Capella, to cover what features are available in which editor on Server, plus a lacking See Also section

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11791] Adjusting the procedure for creating a Vector Search index to be more self-contained.

Small edits to create-type-mapping.

sarahlwelton committed:
Merge branch 'DOC-11791-create-vector-search-index' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11974] Remove unnecessary "size" parameter

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11794] Correcting incorrect table parameter

sarahlwelton committed:
Merge branch 'DOC-11794-vector-search-rest-api' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11789] Removing unnecessary period in search.adoc

Fixing broken link in quick-index-field-options.adoc

sarahlwelton committed:
[DOC-11790] Adding in integrations.adoc link and removing comments

sarahlwelton committed:
Merge branch 'release/7.6' into DOC-11789-vector-search

ElliotFrancisHunter commented on modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:
```diff
@@ -5,14 +5,14 @@
 :underscore: _
 
-Search functions enable you to use full text search (xref:fts:full-text-intro.adoc[FTS]) queries directly within a {sqlpp} query.
+Search functions enable you to use xref:search:search.adoc[full text search (FTS)] queries directly within a {sqlpp} query.
```
Would this not be Full Text Search?

sarahlwelton replied:
Yeah, I suppose we could capitalize.

ElliotFrancisHunter commented on modules/search/pages/child-field-options-reference.adoc:
```diff
@@ -0,0 +1,98 @@
+The `_all` field is a composite field that has the content from multiple fields in an index. 
+It allows searches to query the content of a field without specifying the field's name. 
+
+To include this field in the `_all` field, select *Include in _all field*. 
```
Should that be *Include in _all*  field? As in 'field' is outside of this.

sarahlwelton replied:
Nope, the setting does appear as "Include in _all field" in the UI.

ElliotFrancisHunter replied:
Fair enough 👍🏼

ElliotFrancisHunter commented on modules/search/pages/customize-index.adoc:
```diff
@@ -5,41 +5,58 @@
+* *Dynamic type mappings*: When you do not know the structure of your data fields ahead of time, use a dynamic type mapping to add all available fields from a matching document type to an index.
+For example, you could create a dynamic type mapping to include all documents from the `hotel` collection in your Search index, or include all fields under a JSON object from your document schema.
++
+Configure this type of mapping by selecting a collection in xref:create-quick-index.adoc[the Quick Editor] or by clearing *Only index specified fields* when you xref:create-type-mapping.adoc[].
```
Should this be only index specified fields?

sarahlwelton replied:
Can you elaborate on what you mean?

ElliotFrancisHunter replied:
I am a tad confused why 'Only' is capitalised is all :)

sarahlwelton replied:
Again, matching the appearance in the UI!

ElliotFrancisHunter commented on modules/search/pages/field-data-types-reference.adoc:
```diff
@@ -34,7 +35,31 @@ The field contains geopoint (latitude and longitude) data, represented as either
+For example: 
+----
+{ 
+    "type": "linestring",
```
Is this correct? It's described in CamelCase below and all lower case here

sarahlwelton replied:
Oh good point - it's a mess of consistency even in the examples I was finding. Let me fix that....

ElliotFrancisHunter commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+
+You can develop applications that include: 
+
+* *Similarity search:* Search for documents, products, images, and more that are similar to a given query.
```
Should this be Similarity Search?

sarahlwelton replied:
It's not a proper noun, so that's why I have it lowercased :)

ElliotFrancisHunter reviewed (commented):
Some really good work here! Just a few clarifications for me on some terms but other than that it looks awesome 🙌🏼

sarahlwelton committed:
[DOC-11789] Consistency edits based on peer review

julia-cb commented on modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:
```diff
@@ -118,10 +118,10 @@ The object may contain the following fields:
-| The `index` field may be a string, containing the name of a full text index in the keyspace.
-(This may be a full text index alias, but only if the full text index is in the same keyspace.)
+| The `index` field may be a string, containing the name of a Search index in the keyspace.
+(This may be a Search index alias, but only if the Search index is in the same keyspace.)
```
I'd suggest using "can/might" instead of "may" in docs. "May" can be ambiguous, especially for ESL speakers.

sarahlwelton replied:
I didn't write most of this, but I can change this while I'm in here. I was just tweaking some existing names/phrases.

julia-cb commented on modules/search/pages/create-type-mapping.adoc:
```diff
@@ -28,12 +28,13 @@ To create a type mapping with the Couchbase Server Web Console:
 . (Optional) To only include documents of a specific type from a collection, in the *#* field, add the document type to the end of the collection.
```
```suggestion
. (Optional) To only include documents of a specific type from a collection, add the document type to the end of the collection in the *#* field.
```

sarahlwelton committed:
[DOC-11789] Copying over applicable comments from Capella peer review

julia-cb commented on modules/search/pages/create-child-field.adoc:
```diff
@@ -26,79 +26,16 @@ To add or remove a child field from a Search index with the Couchbase Server Web
-. In the *Field* field, enter the name of a field in your documents that you want to add or remove from the index.
+. [[field]]In the *Field* field, enter the name of a field in your documents that you want to add or remove from the index.
 +
 NOTE: The field must contain either a single value or an array. If the field contains a JSON object, xref:create-child-mapping.adoc[Create a Child Mapping], instead.  
```
```suggestion
NOTE: The field must contain either a single value or an array. If the field contains a JSON object, xref:create-child-mapping.adoc[create a child mapping] instead.  
```

julia-cb replied:
Super nitpicky. I realize you didn't originally write this - I'll focus on the new content you added for the rest of the review :)

sarahlwelton committed:
Apply suggestions from code review

Co-authored-by: Julia B <150837331+julia-cb@users.noreply.github.com>

sarahlwelton committed:
[DOC-11789] Applying suggestion from peer review

sarahlwelton committed:
Merge branch 'DOC-11789-vector-search' of https://github.com/couchbaselabs/docs-devex into DOC-11789-vector-search

julia-cb commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,73 @@
+.. In the *Collection* list, select the collection that contains the documents you want to include in your index. 
+.. Select *Only index specified fields*. 
+.. Click btn:[OK].
+. xref:search:create-child-field.adoc[] on the new collection type mapping with the following settings: 
+.. In the *Field* field, enter the name of the field in your documents that contains your vector embeddings.
++
+Vector embeddings should be an array of numbers. 
+.. In the *Type* list, select *vector*. 
+.. In the *Dimension* field, enter the total number of elements in the array that holds the vector embeddings for your documents.
++ 
+The Search Service supports arrays up to 2048 elements. 
+.. In the *Similarity metric* list, select the method to use to calculate the similarity between search term and Search index vectors.
++ 
+For more information, see xref:search:child-field-options-reference.adoc[].
+.. In the *Optimized for* list, select whether the Search Service should optimize Search queries for accuracy (*recall*) or speed (*latency*).
++
+For more information, see xref:search:child-field-options-reference.adoc[].
+.. Select *Index*.
+.. Click btn:[OK].
```
```suggestion
. xref:search:create-child-field.adoc[] on the new collection type mapping. 
Make sure to include the vector settings *Dimension*, *Similarity metric*, and *Optimized for*.
+
For more information on child field settings, refer to xref:search:child-field-options-reference.adoc[].
```

julia-cb replied:
Just a suggestion - totally up to you. The info already exists on the other pages so it feels like it's just being duplicated in this task page.

sarahlwelton replied:
I'll have to think on this one. 🤔

julia-cb reviewed (approved):
What a beast! Great job, Sarah :)

(Also, I agree with Daniel that "Use Vector Search for AI Applications" is a great title for this new section!!)

sarahlwelton committed:
Update modules/search/pages/create-child-field.adoc

Co-authored-by: Julia B <150837331+julia-cb@users.noreply.github.com>

sarahlwelton committed:
[DOC-11789] Correcting with another peer review comment

sarahlwelton committed:
Merge branch 'DOC-11789-vector-search' of https://github.com/couchbaselabs/docs-devex into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11789] Adding a tip to make the connection between Search and Vector Search clear.

sarahlwelton committed:
[DOC-11789] Applying suggestions from peer review

sarahlwelton committed:
[DOC-11789] Feedback from SME review

sarahlwelton committed:
Merge branch 'DOC-11666-7-6-search' into DOC-11789-vector-search

sarahlwelton committed:
[DOC-11789] Consistency edits and SME reviews

sarahlwelton committed:
[DOC-11789] Adding clarification based on MB-60141

sarahlwelton committed:
[DOC-11789] Add missing "run a search with python" page to vector search + nav.adoc

sarahlwelton committed:
[DOC-11789] Big consistency pass in tandem with Capella to clear up any remaining inconsistencies between files

sarahlwelton committed:
[DOC-11793] Updating Integrations page for LangChain

sarahlwelton committed:
[DOC-11789] Adding attributes for product and UI names to future proof docs updates

Adding proper expansion formatting for variables in REST API endpoints and curl commands

sarahlwelton committed:
[DOC-11789] May -> Can, per peer review

sarahlwelton committed:
[DOC-11789] Correcting error in attribute definitions

Updating integrations.adoc

sarahlwelton committed:
[DOC-11789] Fixing broken links

sarahlwelton committed:
[DOC-11789] Changing integrations.adoc into a partial to include in Capella doc
